### PR TITLE
Added Election Term Metric

### PR DESCRIPTION
**Pull request by meetshah777**
**This Metric will Publish the Only Election term number**

Tests:
_Docker Container Testing_

1. Output of tmp file

```
^election_term
{"current_time":1612872530595}
{"Election_Term":18}$
```
2. Entry in Election term table

```
sqlite> select * from Election_Term;
18.0|18.0|18.0|18.0
```
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ElectionTermCollector.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        public ElectionTermMetrics(long electionTerm) {
+            this.electionTerm = electionTerm;
+        }
+
+        @JsonProperty(AllMetrics.ElectionTermValue.Constants.ELECTION_TERM_VALUE)
+        public long getElectionTerm() {
+            return electionTerm;
+        }
+    }
+}
```
Comment (yu-sun-77) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ElectionTermCollector.java:102]: can we add a newline?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ElectionTermCollector.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        super(SAMPLING_TIME_INTERVAL, "ElectionTermCollector");
+        value = new StringBuilder();
+        this.controller = controller;
+        this.configOverridesWrapper = configOverridesWrapper;
+    }
+
+    @Override
+    public String getMetricsPath(long startTime, String... keysPath) {
+        // throw exception if keysPath.length is not equal to 0
+        if (keysPath.length != KEYS_PATH_LENGTH) {
```
Comment (yu-sun-77) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ElectionTermCollector.java:51]: can we add a comment to provide an example of expected keysPath

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ElectionTermCollectorTests.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        assert metrics.size() == size;
+        if (size != 0) {
+            String[] jsonStrs = metrics.get(0).value.split("\n");
+            assert jsonStrs.length == 2;
+            return jsonStrs[1];
+        } else {
+            return null;
+        }
+    }
+}
```
Comment (yu-sun-77) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ElectionTermCollectorTests.java:125]: can we add newline?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ElectionTermCollector.java (hunk 2 above)
Comment (meetshah777) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ElectionTermCollector.java:51]: KeyPath should be null. Metric path will be `election_term`. Added comment that keyPath length should be 0

**Review by yu-sun-77 — APPROVED** [review]

**Review by yu-sun-77 — CHANGES_REQUESTED** [review]
There are compilation error, can you please check?
https://github.com/opendistro-for-elasticsearch/performance-analyzer/pull/297/checks?check_run_id=2193042070

**Review by yu-sun-77 — APPROVED** [review]

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ElectionTermCollector.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.MetricsConfiguration;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.MetricsProcessor;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.PerformanceAnalyzerMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.metrics.ExceptionsAndErrors;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.metrics.WriterMetrics;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
```
Comment (khushbr) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ElectionTermCollector.java:31]: Let's add a docString here about what this collector does.

Comment (meetshah777) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ElectionTermCollector.java:31]: done

**Review by khushbr — APPROVED** [review]

**Review by yu-sun-77 — APPROVED** [review]

**Pull request merged**